consensus/src/rand/rand_gen/rand_manager.rs

```diff
@@ -213,23 +213,38 @@ impl<
         DropGuard::new(abort_handle)
     }
 
-    fn broadcast_aug_data(&self) -> DropGuard {
-        let data = D::generate(&self.config);
+    async fn broadcast_aug_data(&mut self) -> CertifiedAugData<D> {
+        if let Some(certified_data) = self.aug_data_store.get_my_certified_aug_data() {
+            info!("[RandManager] Already have certified aug data");
+            return certified_data;
+        }
+        let data = self
+            .aug_data_store
+            .get_my_aug_data()
+            .unwrap_or_else(|| D::generate(&self.config));
+        // Add it synchronously to avoid race that it sends to others but panics before it persists locally.
+        self.aug_data_store
+            .add_aug_data(data.clone())
```

Comment:
If `aug_data_store` already has data, will add fail?

Reply:
no as long as it’s the same data